flow-model-generator/modules/flow-model-generator-core/src/main/java/io/ballerina/flowmodelgenerator/core/expressioneditor/semantictokens/SemanticTokenVisitor.java

```diff
@@ -89,7 +91,25 @@ public void visit(SimpleNameReferenceNode simpleNameReferenceNode) {
 
     @Override
     public void visit(FieldAccessExpressionNode fieldAccessExpressionNode) {
-        // Mark the field name as PROPERTY
+        // Find the leftmost node
+        ExpressionNode expression = fieldAccessExpressionNode.expression();
+        while (expression.kind() == SyntaxKind.FIELD_ACCESS) {
+            expression = ((FieldAccessExpressionNode) expression).expression();
+        }
+
+        // Mark the field as variable if the expression is a variable reference
+        if (expression instanceof SimpleNameReferenceNode nameReferenceNode) {
```

Comment:
Wouldn't the recommended way is to check the syntax node `kind`, instead of doing `instance of` checks?
```suggestion
        if (expression instanceof SimpleNameReferenceNode nameReferenceNode) {
```

Reply:
Ack.